File: 2022/day2/elixir/part2.exs
defmodule MyModule do
  @lose "X"
  @draw "Y"
  @win "Z"
  def main do
    m = %{
      ["A", @lose] => 3 + 0,
      ["B", @lose] => 1 + 0,
      ["C", @lose] => 2 + 0,
      ["A", @draw] => 1 + 3,
      ["B", @draw] => 2 + 3,
      ["C", @draw] => 3 + 3,
      ["A", @win] => 2 + 6,
      ["B", @win] => 3 + 6,
      ["C", @win] => 1 + 6
    }

    contents =
      File.read!("../input.txt") |> String.split("\r\n") |> Enum.filter(fn line -> line != "" end)

    total =
      Enum.reduce(contents, 0, fn line, acc ->
        [opponent, player] = String.split(line, " ")
        result = Map.get(m, [opponent, player])
        acc + result
      end)

    IO.puts(total)
  end
end

MyModule.main()

File: 2022/day2/cpp/part2.cpp
#include <fstream>
#include <iostream>

using namespace std;

int main() {
    char x, z;
    int score{};
    ifstream f("input.txt");
    while (f >> x) {
        f >> z;
        if (z == 'X') {
            if (x == 'A')
                score += 3;
            if (x == 'B')
                score += 1;
            if (x == 'C')
                score += 2;
        }
        if (z == 'Y') {
            if (x == 'A')
                score += 4;
            if (x == 'B')
                score += 5;
            if (x == 'C')
                score += 6;
        }
        if (z == 'Z') {
            if (x == 'A')
                score += 8;
            if (x == 'B')
                score += 9;
            if (x == 'C')
                score += 7;
        }
    }
    cout << score << endl;
}